android/src/main/java/io/ably/flutter/plugin/util/NotificationUtilities.java

```diff
@@ -0,0 +1,191 @@
+   */
+  public static void showNotification(Context context, RemoteMessage remoteMessage, Bundle intentExtras) {
+    RemoteMessage.Notification notification = remoteMessage.getNotification();
+    Notification localNotification = NotificationUtilities.createLocalNotification(context, notification);
+    Bundle manifestMetadata = getManifestMetadata(context.getPackageManager(), context.getPackageName());
+    NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+    // Create a unique notification ID so notifications don't share the same ID, as per https://stackoverflow.com/a/28251192/7365866
+    int notificationId = (int) ((new Date().getTime() / 1000L) % Integer.MAX_VALUE);
+    NotificationParams params = new NotificationParams(intentExtras);
+    CommonNotificationBuilder.DisplayNotificationInfo notificationInfo = CommonNotificationBuilder.createNotificationInfo(
+        context,
+        context.getPackageName(),
+        params,
+        getOrCreateChannel(context, notification.getChannelId()),
+        context.getResources(),
+        context.getPackageManager(),
+        manifestMetadata
+        );
+    notificationManager.notify(notificationInfo.tag, notificationInfo.id, notificationInfo.notificationBuilder.build());
+  }
+
+  /**
+   * The [firebase-android-sdk](https://github.com/firebase/firebase-android-sdk) only shows
+   * notifications when the app is not in the foreground. iOS, this is configurable. To make
+   * this behaviour consistent between Android and iOS, we allow users to configure notifications
+   * when the app is in the foreground, by creating a notification for them, which closely mimics
+   * what firebase-android-sdk creates when the app is not in the foreground.
+   *
+   * @param context
+   * @param notification
+   * @return
+   */
+  public static Notification createLocalNotification(Context context, RemoteMessage.Notification notification) {
+    @Nullable String notificationChannelId = NotificationUtilities.getOrCreateChannel(context, notification.getChannelId());
+    NotificationCompat.Builder builder = new NotificationCompat.Builder(context, notificationChannelId)
+        .setContentTitle(notification.getTitle())
+        .setContentText(notification.getBody())
+        .setSmallIcon(context.getApplicationContext().getApplicationInfo().icon)
+        .setPriority(NotificationCompat.PRIORITY_DEFAULT);
+    if (notification.getImageUrl() != null) {
+      downloadImage(notification.getImageUrl(), (bitmap) -> {
+        builder.setLargeIcon(bitmap);
+        builder.setStyle(new NotificationCompat.BigPictureStyle()
+            .bigPicture(bitmap)
+            .bigLargeIcon(null));
+        builder.build();
+      });
+    }
+    return builder.build();
+  }
+
+
+  private interface DownloadImageCallback {
+    void finish(Bitmap bitmap);
+  }
+
+  private static void downloadImage(Uri imageUri, DownloadImageCallback callback) {
```

Comment:
In Android you normally cannot perform a network download in the main thread. This method probably should be commented to warn users about its usage

Reply:
Also should move to another class as it is not directly related to notifications

Reply:
Most of this file contents is taken from Firebase, and I wanted to put in the same file. It's a private method. Now, I'm even thinking about removing this entire file actually before this PR merges.

Reply:
Repeating a comment from https://github.com/ably/ably-flutter/pull/166#discussion_r716573968:
> This file was deleted, because the entire functionality of this file IMHO should not be in the project. The functionality: Firebase Android SDK does not show notifications when the app is in the foreground, but on iOS it is customisable. I copied some code from Firebase and modified it to allow users to configure it.
>
> I have removed this now: c5070b3